### DISCUSSION TITLE
[Item 7] 배열 원소의 참조를 해제하는 경우

### DISCUSSION THREAD

**Discussion by nusuy** [GDSC-SWU/2023-JAVA-Study#150] [Q&A] [answered]
p.37에 따르면, 참조를 다 사용했을 때는 null을 대입하여 참조를 해제하면 된다고 한다.
하지만, 배열의 원소값으로 '값이 없음'을 의미하고자 null을 대입하는 경우가 있을 텐데, 이 경우와 참조를 해제하는 경우를 어떻게 구분할 수 있을까?

**Comment by yerim123456** [comment, marked as answer]
'값이 없음'을 의미하는 null을 대입한 경우, 해당 레퍼런스 변수가 참조하는 객체가 없다는 것일 뿐입니다. 그렇기에 참조를 해제하는 경우와 null을 대입하는 경우를 구분할 필요가 없을 것 같습니다. 만약, 이미 참조하는 객체가 있는 경우에는 참조를 해제하는 의미와 동시에 값이 없다는 의미를 갖게 됩니다. 만약, 처음 정의를 null로 했을 경우에는 그저 값이 없다는 의미만 갖게 된다고 생각합니다.
